go/vt/vtgate/evalengine/internal/json/parser_test.go

```diff
@@ -0,0 +1,773 @@
+/*
+Copyright 2018 Aliaksandr Valialkin
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package json
+
+import (
+	"testing"
+
+	"vitess.io/vitess/go/hack"
+)
+
+func TestParseRawNumber(t *testing.T) {
+	t.Run("success", func(t *testing.T) {
+		f := func(s, expectedRN, expectedTail string) {
+			t.Helper()
+
+			rn, tail, err := parseRawNumber(s)
+			if err != nil {
+				t.Fatalf("unexpected error: %s", err)
+			}
+			if rn != expectedRN {
+				t.Fatalf("unexpected raw number; got %q; want %q", rn, expectedRN)
+			}
+			if tail != expectedTail {
+				t.Fatalf("unexpected tail; got %q; want %q", tail, expectedTail)
+			}
```

Comment:
nit: mostly codebase has use of `github.com/stretchr/testify/require` for assertion

Reply:
@harshit-gangal: these are vendored tests, so I've opted to leave them using the same assertions as the original so they can be upgraded without hassle. You can see that the new tests I've introduced for the JSON package do use `assert` and `require`.